File: implementations/php/src/BlockView.php
<?php

declare(strict_types=1);

namespace Kduma\PCF;

/**
 * One table block read from disk: its absolute offset, its parsed
 * {@see TableBlockHeader} (including table_hash and next_table_offset), and its
 * {@see PartitionEntry} list.
 *
 * This is a read-only view returned by {@see Container::readBlockAt()}. It lets
 * code layered on PCF group blocks, inspect each block's table_hash, and follow
 * non-default next_table_offset chains, instead of {@see Container::entries()}
 * which flattens the whole chain.
 */
final class BlockView
{
    /**
     * @param PartitionEntry[] $entries the block's entries, in stored order
     */
    public function __construct(
        public int $offset,
        public TableBlockHeader $header,
        public array $entries,
    ) {
    }
}

File: implementations/dotnet/src/Pcf/BlockView.cs
using System.Collections.Generic;

namespace Pcf;

/// <summary>
/// One table block read from disk: its absolute <see cref="Offset"/>, its parsed
/// <see cref="TableBlockHeader"/> (including <c>table_hash</c> and
/// <c>next_table_offset</c>), and its <see cref="PartitionEntry"/> list.
/// </summary>
/// <remarks>
/// This is a read-only view returned by <see cref="Container.ReadBlockAt"/>. It
/// exists so that profiles layered on PCF (which must group blocks, inspect each
/// block's <c>table_hash</c>, and follow non-default <c>next_table_offset</c>
/// chains) can reuse PCF's block parsing rather than re-decoding raw bytes. It
/// plays no part in the writer's in-memory bookkeeping.
/// </remarks>
public sealed class BlockView
{
    /// <summary>Absolute file offset of the table block.</summary>
    public ulong Offset { get; }

    /// <summary>Parsed 74-byte block header.</summary>
    public TableBlockHeader Header { get; }

    /// <summary>The block's entries, in stored order.</summary>
    public IReadOnlyList<PartitionEntry> Entries { get; }

    /// <summary>Create a block view.</summary>
    public BlockView(ulong offset, TableBlockHeader header, IReadOnlyList<PartitionEntry> entries)
    {
        Offset = offset;
        Header = header;
        Entries = entries;
    }
}
